components/ipfs/ipfs_utils.cc

```diff
@@ -445,4 +445,47 @@ std::string GetRegistryDomainFromIPNS(const GURL& url) {
       cid, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
 }
 
+// gateway.io/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy ->
+// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
+// bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy.ipfs.gateway.io ->
+// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
+absl::optional<GURL> TranslateToCurrentGatewayUrl(const GURL& url) {
+  if (!url.is_valid() || !url.SchemeIsHTTPOrHTTPS()) {
+    return absl::nullopt;
+  }
+
+  std::vector<std::string> host_parts = base::SplitStringUsingSubstr(
+      url.host(), ".", base::KEEP_WHITESPACE, base::SPLIT_WANT_ALL);
+
+  if (host_parts.size() > 2 && IsValidCID(host_parts.at(0)) &&
+      host_parts.at(1) == "ipfs") {
+    GURL final_url = GURL("ipfs://" + host_parts.at(0) + url.path());
```

Comment:
you're redirecting http/https to an ipfs url? I don't think that is the correct way to handle this, you should be using url rewriting to display ipfs as the virtual url

Reply:
this is the reverse of what we do when someone enters `ipfs://` in the urlbar

Reply:
We are checking the response header at the moment

Reply:
so this redirects any gateway url? I don't think we should be doing that. If I explicitly enter an ipfs url that uses a different gateway than what I have configured, it should not redirect to a different gateway. I thought this was only to display ipfs for any urls that matched the configured gateway which can be done with the rewriter.

Reply:
> so this redirects any gateway url? I don't think we should be doing that. If I explicitly enter an ipfs url that uses a different gateway than what I have configured, it should not redirect to my configured gateway. I thought this was only to display ipfs for any urls that matched the configured gateway which can be done with the rewriter.

Nope, this is about reusing configured gateway to load IPFS resources.
> Automatically uses the configured gateway for IPFS resolutions when an IPFS gateway resource is encountered.
As for  the manual url input i'll discuss it, maybe this should be exclusion.